### ISSUE TITLE
LaTeX exponential notation flaws

### ISSUE THREAD

**Issue by misharash** [astropy/astropy#12571]
### Description
LaTeX exponential notation formatting doesn't handle value of 1 properly and wrongly puts units in the exponent.

### Expected behavior
Not entirely clear how to format value of 1, but just `1` or `10^0` are both decent choices. The first seems more consistent, given that e.g. 1.5 is formatted as `1.5`.
Units should not be in exponent.

### Actual behavior

1 without units is formatted as an empty string. 1 with units, e.g. 1 cm, is missing power index in the exponent.
Units enter the exponent.

This is confusing.

### Steps to Reproduce

```python
import astropy.units as u
print(u.format.Latex.format_exponential_notation(1, ".1e"))
print(u.format.Latex.format_exponential_notation(1*u.cm, ".1e"))
print(u.format.Latex.format_exponential_notation(1.5*u.cm, ".1e"))
print(u.format.Latex.format_exponential_notation(.1*u.cm, ".1e"))
print(u.format.Latex.format_exponential_notation(.15*u.cm, ".1e"))
```
Output:
```

10^{ cm}
1.5 \times 10^{ cm}
10^{-1 cm}
1.5 \times 10^{-1 cm}
```
Expected output:
```
1
1 cm
1.5 cm
10^{-1} cm
1.5 \times 10^{-1} cm
```

### System Details

Linux-5.15.6-zen2-1-zen-x86_64-with-glibc2.33
Python 3.9.9 (main, Nov 20 2021, 21:30:06) 
[GCC 11.1.0]
Numpy 1.21.3
pyerfa 2.0.0.1
astropy 5.0
Scipy 1.7.3
Matplotlib 3.5.0

Also checked that it's identical with astropy 5.1.dev231+g584ae2e92

### P.S.
I haven't found any other way other than `astropy.units.format.Latex.format_exponential_notation` that could produce LaTeX exponential notation with `X \times 10^{Y}` and not `XeY`. And unfortunately it does not currently format units equally nicely (e.g. `Msun` becomes `solMass` and not `M_{\odot}`). `Quantity.to_string(format="latex")` does nice units, but I can't ask for exponential notation manually, values between 1e-4 and 1e7 do not use it. Please point out if there is an alternative.

**Comment by mhvk** [comment]
@misharash - thanks for the report! I think there are two issues here, one that perhaps `utils.split_mantissa_exponent` does a bit too much, by removing a mantissa of exactly 1, and the other that one cannot set the range within which exponential notation is used by `Quantity.to_string(format='latex')`. (The `format_exponential_notation` class method is not meant to taken `Quantity`, so I think that part is OK.)

If I understand correctly, the latter is the reason for your issue, and I agree it would be good to allow more flexibility!

The underlying reason why values between 1e-4 and 1e7 are not typeset as you want, is https://github.com/astropy/astropy/blob/bed1f69cc1510251bd2c11ee079ce0f80d79bec3/astropy/units/quantity.py#L1367

The way the code is currently written, it assumes a `g` format, which does that. 

It should not be too hard to change this, but it is probably good to think through a bit exactly how to allow more freedom. Perhaps allowing the user to pass in `format_spec`? Or exposing the `formatter` from `array2string`?

**Comment by mhvk** [comment]
p.s. Just looking at the code, I noticed a small bug, so this was useful already! See gh-12573

**Comment by misharash** [comment]
@mhvk - thanks for the prompt response! I see, I missed that `format_exponential_notation` is not supposed to take `Quantity`, only simple numbers.
Personally I'd prefer to be able to just pass `format_spec`, since I'm not really familiar with the other way you propose. Although I'm not sure it's representative of all the community.

**Comment by mhvk** [comment]
> I addressed the first part in #15923. Now, the remaining part reads more like a feature request, so I'd suggest updating labels on this issue.

I changed labels, but then reconsidered: probably easier to have a new issue - see #15933

**Comment by mhvk** [comment]
p.s. Thanks for raising this issue, @misharash! At least it is now partially addressed.
